ApplicationLibCode/UserInterface/RiuPlotWidget.cpp

```diff
@@ -217,6 +226,80 @@ void RiuPlotWidget::updateOverlayFrameLayout()
     }
 }
 
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+bool RiuPlotWidget::handleDragDropEvent( QEvent* event )
+{
+    if ( !event ) return false;
+
+    if ( event->type() == QEvent::DragEnter )
+    {
+        auto dragEnterEvent = dynamic_cast<QDragEnterEvent*>( event );
+        if ( dragEnterEvent )
+        {
+            dragEnterEvent->acceptProposedAction();
+
+            return true;
+        }
+    }
+
+    const MimeDataWithIndexes* mimeData = nullptr;
+
+    if ( event->type() == QEvent::Drop )
+    {
+        // These drop events come from Qwt
+        auto dropEvent = dynamic_cast<QDropEvent*>( event );
+        if ( dropEvent )
+        {
+            mimeData = qobject_cast<const MimeDataWithIndexes*>( dropEvent->mimeData() );
+
+            dropEvent->acceptProposedAction();
+        }
+    }
+
+    if ( event->type() == QEvent::GraphicsSceneDrop )
+    {
+        // These drop events come from QtChart
+        auto dropEvent = dynamic_cast<QGraphicsSceneDragDropEvent*>( event );
+        if ( dropEvent )
+        {
+            mimeData = qobject_cast<const MimeDataWithIndexes*>( dropEvent->mimeData() );
+
+            dropEvent->acceptProposedAction();
+        }
+    }
```

Comment:
Maybe combine these two in to one "if"? Cast to QDropEvent* since both mimeData and acceptProposedAction is available there.

Reply:
Not possible, the common base class of QDropEvent and QGraphicsSceneDragDropEvent is QEvent